_pages/algorithm-development/languages/python.md

```diff
@@ -10,57 +10,90 @@ image:
 
-Go ahead and remove the boilerplate code below that's inside the apply() function because we'll be writing a different algorithm in this tutorial:
+{% highlight python %}
+from Algorithmia import ADK
 
-<img src="{{site.cdnurl}}{{site.baseurl}}/images/post_images/algo_dev_lang/algorithm_console_python.png" alt="Algorithm console Python" class="screenshot">
+def apply(input):
+    return "hello {}".format(str(input))
 
+algorithm = ADK(apply)
+algorithm.init("Algorithmia")
+{% endhighlight %}
 
-## Managing Dependencies
+When executed on the Algorithmia platform and providing the string "HAL 9000" as an input, this algorithm will output "hello HAL 9000". If executed locally, the algorithm will print `hello Algorithmia` to `stdout` instead.
 
-Algorithmia supports adding 3rd party dependencies via the <a href="https://pypi.python.org/pypi">Python Package Index (PyPI)</a> using a requirements.txt file.
+## Loaded State
 
-On the algorithm editor page there is a button on the top right that says "Dependencies". Click that button and you'll see a modal window:
+Developing with the ADK allows you to make use of an optional `load()` function for preparing an algorithm for runtime operations. Using `load()`, we can load state into memory prior to a function's execution and then pass that data to `apply()` by adding an additional `globals` parameter in the `apply()` function. 
 
-<img src="{{site.cdnurl}}{{site.baseurl}}/images/post_images/algo_dev_lang/dependencies_python.png" alt="Python Dependency File" class="screenshot img-md">
+Let's add a `load()` function to the Hello World example we just created:
 
-If you have any dependencies you can add them by typing in the package name to the `requirements.txt` file.
+{% highlight python %}
+from Algorithmia import ADK
 
-This guide won't depend on any external dependencies so you can close the dependencies window.
+def apply(input, globals):
+    return "hello {} {}".format(str(input), str(globals['payload']))
 
-If you do add dependencies, you will still need to import those packages via the import statement to your algorithm file as you would do for any Python script.
-{: .notice-info}
+def load():
+    globals = {}
+    globals['payload'] = "Loading has been completed."
+    return globals
+
+algorithm = ADK(apply, load)
+algorithm.init("Algorithmia")
+{% endhighlight %}
+
+In our load function we've created a `globals` object and added a key called `payloads` with a string value `Loading has been completed.`; we then return the `globals` object, which will be passed as input to the algorithm's `apply()` function, where we use that value as part of the algorithm's output. Executing the algorithm locally will result in `hello Algorithmia Loading has been completed.` being printed to `stdout`.
```

Comment:
Again, I'm feeling as the "why" is missing in these docs

Reply:
which part of this should have a why you think? This allows your apply function to be able to explicitly use and interact with files and logic that was defined during the load function - the goal here is to avoid using global state as much as possible, as that has caused issues for customers in the past; and also is a code-smell in most more mature workflows.